### PR TITLE
Adding Parameter Sweep for Two Hidden Layer Model / Reorganizing nbconverted Scripts

### PR DESCRIPTION
**Pull request by gwaygenomics**
A lot of intermediate files being added in the PR in commit 9997c6d. These files are the result of a parameter sweep of a two hidden layer model (5000 -> 100 -> 100 -> 100 -> 5000). Commit 2b61e96 adds the compiled file with git LFS.

The main file changes include adding options to the parameter sweep script (cd77571) and adding a script that actually trains the models (7abeb7b). This script is very similar to the tybalt notebook.

### CODE REVIEW

File: scripts/vae_pancancer.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+# Set hyper parameters
+learning_rate = float(args.learning_rate)
+batch_size = int(args.batch_size)
+epochs = int(args.epochs)
+kappa = float(args.kappa)
+depth = int(args.depth)
+first_layer = int(args.first_layer)
+output_filename = args.output_filename
+
+original_dim = 5000
```
Comment (jaclyn-taroni) [inline scripts/vae_pancancer.py]: Might change if no. genes based on MAD filter changes?

File: scripts/vae_pancancer.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+rnaseq_train_df = rnaseq_df.drop(rnaseq_test_df.index)
+
+# Input place holder for RNAseq data with specific input size
+rnaseq_input = Input(shape=(original_dim, ))
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~
+# ENCODER
+# ~~~~~~~~~~~~~~~~~~~~~~
+# Input layer is compressed into a mean and log variance vector of size
```
Comment (jaclyn-taroni) [inline scripts/vae_pancancer.py]: Does this need to be updated to take into account the two hidden layers? Hard to tell

File: scripts/vae_pancancer.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+# return the encoded and randomly sampled z vector
+# Takes two keras layers as input to the custom sampling function layer with a
+# latent_dim` output
+z = Lambda(sampling,
+           output_shape=(z_shape, ))([z_mean_encoded, z_log_var_encoded])
+
+# ~~~~~~~~~~~~~~~~~~~~~~
+# DECODER
+# ~~~~~~~~~~~~~~~~~~~~~~
+# The decoding layer is much simpler with a single layer glorot uniform
```
Comment (jaclyn-taroni) [inline scripts/vae_pancancer.py]: Same comment as above

**Comment by danich1** [comment]
I agree with @jaclyn-taroni comments. Just curious as to why a completely new script instead of modifying `tybalt_vae.py`?

**Comment by jaclyn-taroni** [comment]
I have the same question as @danich1, that's what I was trying to get at. I thought it might be for notebook reasons?

File: scripts/vae_pancancer.py (hunk 1 above)
Comment (gwaygenomics) [reply scripts/vae_pancancer.py]: :+1:

**Comment by gwaygenomics** [comment]
> Just curious as to why a completely new script instead of modifying tybalt_vae.py?

> I thought it might be for notebook reasons?

Precisely. I think it would be best to keep the nbconverted scripts separate from scripts run consistently. If they are not separate, when a notebook changes, it would get messy quick.

Do you think the repo would be better organized with a `scripts/nbcovert/` folder storing these scripts instead?

**Comment by danich1** [comment]
I concur on separating out the nbconverted scripts. Better to separate them off to avoid confusion about which is suppose to be a converted script and what is suppose to be a run-consistent script.

**Comment by gwaygenomics** [comment]
thanks @danich1 and @jaclyn-taroni - I believe b895ece addresses your concerns. I also believe that some portion of concern is regarding redundancy and I definitely agree. 

Portions of this code base are getting entirely too redundant, and this realization prompts refactorization (see #74 for more details).

**Comment by gwaygenomics** [comment]
> I concur on separating out the nbconverted scripts. Better to separate them off to avoid confusion about which is suppose to be a converted script and what is suppose to be a run-consistent script.

Sounds good. in the next commit, inside this PR (I will also rename the PR). I will reorganize nbconverted scripts.

**Comment by gwaygenomics** [comment]
thanks @danich1 and @jaclyn-taroni - ready for review again

**Comment by jaclyn-taroni** [comment]
@gwaygenomics - Does your output change when setting `kernel_initializer='glorot_uniform'` in your decoder in [`b895ece`](https://github.com/greenelab/tybalt/pull/71/commits/b895ecea5ea7988e0e856a41a25341d8dbdc6df5)? If so, should probably update those files as well here

**Comment by gwaygenomics** [comment]
> Does your output change when setting kernel_initializer='glorot_uniform' in your decoder in b895ece?

Good catch. It does not change since [`glorot_uniform` is the default](https://keras.io/layers/core/). I add it here for consistency and to remove those concerns.

**Pull request merged**